tests/integrationv2/processes.py

```diff
@@ -116,6 +116,14 @@ def _communicate(
         send_with_newline=False,
         timeout=None,
     ):
+        print(
+            f"{self.name}: Handling process IO: \n"
+            f"    wait_for_marker: {self.wait_for_marker} \n"
+            f"    send_markers: {send_marker_list} \n"
+            f"    close_marker: {close_marker} \n"
+            f"    kill_marker: {kill_marker}"
```

Comment:
While debugging flaky tests, I learned (relearned?) that _communicate is actually called multiple times. It can be called once for wait_for_marker, once for the main test IO, and once if the test times out in order to read any remaining output. It behaves differently every time because different markers are passed to it. Our current logging doesn't communicate the multiple calls, which makes it very confusing if _communicate suddenly seems to have no wait_for_marker or no send_marker.